### PR TITLE
don't gossip data back to the node we rcvd it from [#1227]

### PR DESCRIPTION
**Pull request by mynameisdaniil**
Not sure how to go about testing it yet.

### CODE REVIEW

File: fluffy/eth_data/history_data_seeding.nim [hunk 1]
```diff
@@ -60,7 +60,7 @@ proc propagateEpochAccumulator*(
       encodedAccumulator
     )
     discard await p.neighborhoodGossip(
-      ContentKeysList(@[encKey]), @[encodedAccumulator])
+      ContentKeysList(nodeId, @[encKey]), @[encodedAccumulator])
```
Comment (mynameisdaniil) [inline fluffy/eth_data/history_data_seeding.nim:63]: I'm not sure what `nodeId` should it be in this case. Should it be `nodeId` of the current node? What would be a better way to get it in this part of the code?

**Comment by kdeme** [comment]
Looks good. Testing this is not so convenient indeed. It probably would require using some individual functions of the portal protocol and not the full implementation. Another way could be perhaps by accessing the metrics, but again, not so convenient / conventional

File: fluffy/network/wire/portal_protocol.nim [hunk 2]
```diff
@@ -1128,10 +1128,11 @@ proc getNClosestNodesWithRadius*(
 
 proc neighborhoodGossip*(
     p: PortalProtocol,
+    srcNodeId: Opt[NodeId],
     contentKeys: ContentKeysList,
     content: seq[seq[byte]]): Future[int] {.async.} =
   ## Returns number of peers to which content was gossiped
-
+
```
Comment (kdeme) [inline fluffy/network/wire/portal_protocol.nim:1135]: Remove whitespace

File: fluffy/eth_data/history_data_seeding.nim [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import
   std/[strformat, os],
   stew/results, chronos, chronicles,
+  eth/p2p/discoveryv5/protocol,
```
Comment (kdeme) [inline fluffy/eth_data/history_data_seeding.nim:13]: What is this import needed for? Is it for `NodeId`?

If so, I think it is better to add `node` module to the export list of `portal_protocol.nim` as it gets added as parameter in a public API call.

**Pull request merged**